_tests/test_validate.py

```diff
@@ -165,38 +169,160 @@ def test_checkDescriptionMatches_invalid(self):
+			validate.checkAddonId(self.manifest, VALID_SUBMISSION_JSON_FILE, self.submissionData)
 		)
 		self.assertEqual(errors, [])
 
-	def test_checkNameMatchesPath_invalid(self):
+	def test_checkAddonId_invalidPath(self):
+		""" Error when submission path does not include correct addon ID
+
+		Manifest 'name' considered source of truth for addonID
+		Must match:
+		- Submission file name '<addonID>/<version>.json'
+		- `addonId` within the submission JSON data
+		"""
 		filename = os.path.join(TOP_DIR, "invalid")
 		errors = list(
-			validate.checkNameMatchesPath(self.manifest, filename)
+			validate.checkAddonId(self.manifest, filename, self.submissionData)
 		)
 		expectedErrorMessage = validate.ValidationErrorMessage.SUBMISSION_DIR_ADDON_NAME.value
 		self.assertEqual(
 			errors,
 			[expectedErrorMessage.format(self.manifest['name'])]
 		)
 
-	def test_checkVersionMatchesFilename_valid(self):
+	def test_checkAddonId_invalidJSONData(self):
+		""" Error when submission does not include correct addonId
+
+		Manifest 'name' considered source of truth for addonID
+		Must match:
+		- Submission file name '<addonID>/<version>.json'
+		- `addonId` within the submission JSON data
+		"""
+		invalidID = "invalid"
+		self.submissionData['addonId'] = invalidID
+		errors = list(
+			validate.checkAddonId(self.manifest, VALID_SUBMISSION_JSON_FILE, self.submissionData)
+		)
+		expectedErrorMessage = validate.ValidationErrorMessage.ID.value
+		self.assertEqual(
+			[expectedErrorMessage.format(VALID_ADDON_ID, invalidID)],
+			errors
+		)
+
+	def test_checkAddonId_invalidJSONDataAndPath(self):
+		""" Error when submission does not include correct addonId and file path does not include the addonID
+
+		Manifest 'name' considered source of truth for addonID
+		Must match:
+		- Submission file name '<addonID>/<version>.json'
+		- `addonId` within the submission JSON data
+		"""
+		expectedAddonId = "valid"
+		self.manifest['name'] = expectedAddonId
+		errors = list(
+			validate.checkAddonId(self.manifest, VALID_SUBMISSION_JSON_FILE, self.submissionData)
+		)
+		pathMessage = validate.ValidationErrorMessage.SUBMISSION_DIR_ADDON_NAME.value
+		addonIdMessage = validate.ValidationErrorMessage.ID.value
+		self.assertEqual(
+			[
+				pathMessage.format(expectedAddonId),
+				addonIdMessage.format(expectedAddonId, "fake"),
+			],
+			errors
+		)
+
+	def test_checkVersion_valid(self):
+		"""No error when manifest version, submission file name, and submission contents all agree.
+
+		Manifest considered source of truth.
+		Must match:
+		- Submission file name '<addonID>/<version>.json'
+		- `addonVersionField` within the submission JSON data
+		"""
 		errors = list(
-			validate.checkVersionMatchesFilename(self.manifest, VALID_SUBMISSION_JSON_FILE)
+			validate.checkVersions(self.manifest, VALID_SUBMISSION_JSON_FILE, self.submissionData)
 		)
 		self.assertEqual(errors, [])
 
 	def test_checkVersionMatchesFilename_invalid(self):
```

Comment:
Should the function be renamed to something like the following? Just to be consistent.
```suggestion
	def test_checkVersionMatches_invalidFilename(self):
```